roles/sap_general_preconfigure/tasks/main.yml

```diff
@@ -52,45 +61,75 @@
     sap_general_preconfigure_2772999_09: "{{ (sap_preconfigure_2772999_09 | d(sap_general_preconfigure_2772999_09)) | d(false) }}"
     sap_general_preconfigure_2772999_10: "{{ (sap_preconfigure_2772999_10 | d(sap_general_preconfigure_2772999_10)) | d(false) }}"
   when: ansible_facts['distribution'] in ['RedHat']
+  tags:
+    - sap_general_preconfigure_installation
+    - sap_general_preconfigure_configuration
 
 - name: Rename sap_preconfigure_db_group_name if defined
   ansible.builtin.set_fact:
     sap_general_preconfigure_db_group_name: "{{ sap_preconfigure_db_group_name | d(sap_general_preconfigure_db_group_name) }}"
   when: sap_preconfigure_db_group_name is defined or sap_general_preconfigure_db_group_name is defined
+  tags:
+    - sap_general_preconfigure_installation
+    - sap_general_preconfigure_configuration
 
 - name: Set filename prefix to empty string if role is run in normal mode
   ansible.builtin.set_fact:
     __sap_general_preconfigure_fact_assert_filename_prefix: ""
   when: not sap_general_preconfigure_assert | d(false)
+  tags:
+    - sap_general_preconfigure_installation
+    - sap_general_preconfigure_configuration
 
 - name: Prepend filename with assert string if role is run in assert mode
   ansible.builtin.set_fact:
     __sap_general_preconfigure_fact_assert_filename_prefix: "assert-"
   when: sap_general_preconfigure_assert | d(false)
+  tags:
+    - sap_general_preconfigure_installation
+    - sap_general_preconfigure_configuration
 
 # required for installation and configuration tasks:
 - name: Gather package facts
   ansible.builtin.package_facts:
+  tags:
+    - sap_general_preconfigure_installation
 
 - name: Include tasks from 'installation.yml'
-  ansible.builtin.include_tasks: '{{ item }}/{{ __sap_general_preconfigure_fact_assert_filename_prefix }}installation.yml'
+  ansible.builtin.include_tasks:
+    file: '{{ item }}/{{ __sap_general_preconfigure_fact_assert_filename_prefix }}installation.yml'
+    apply:
+      tags: sap_general_preconfigure_installation
```

Comment:
There is already a tag in this task, why adding the same one?

Reply:
This code will apply a tag to all tasks of the included file(s). And without the tags: line below, in the same task, this current task will not be run when using the tag. See also: https://docs.ansible.com/ansible/latest/collections/ansible/builtin/include_tasks_module.html .

Reply:
LGTM